### PR TITLE
add ripemd160 and shake256

### PR DESCRIPTION
**Pull request by la-luo**
## Description of changes in release / Impact of release:
(insert text here)

## Documentation
(insert text here)

### Is this a breaking change?
- [ ] Yes
- [ ] No

### If you answered Yes then describe why is it so
(insert text here if applicable)

### Is there a way to disable the change?
- [ ] Use previous release
- [ ] Use a feature flag
- [ ] No

#### Additional details go here
(insert text here if applicable)

### CODE REVIEW

**Review by aslepakurov — APPROVED** [review]

File: larky/src/main/resources/vendor/Crypto/Hash/Poly1305.star [hunk 1]
```diff
@@ -65,6 +94,21 @@ def Poly1305_MAC(object):
...
         """
+
+    def __init__(r, s, data):
+        if len(r) != 16:
+            fail("ValueError: Parameter r is not 16 bytes long")
+        if len(s) != 16:
+            fail("ValueError: Parameter s is not 16 bytes long")
+
+        self._mac_tag = None 
+        self.digest_size = 16
```
Comment (mahmoudimus) [inline larky/src/main/resources/vendor/Crypto/Hash/Poly1305.star:105]: @la-luo @aslepakurov please return self here.

File: larky/src/main/resources/vendor/Crypto/Hash/RIPEMD160.star [hunk 2]
```diff
@@ -15,20 +28,41 @@ def RIPEMD160Hash(object):
         :vartype digest_size: integer
     
     """
-    def __init__(self, data=None):
+    self = larky.mutablestruct(__class__='RIPEMD160Hash')
```
Comment (mahmoudimus) [inline larky/src/main/resources/vendor/Crypto/Hash/RIPEMD160.star:31]: this should be: `larky.mutablestruct(__name__='RIPEMD160Hash', __class__=RIPEMD160Hash)`

File: larky/src/main/resources/vendor/Crypto/Hash/SHAKE128.star [hunk 3]
```diff
@@ -43,7 +43,7 @@ def SHAKE128_XOF(data=None):
     """
 
     def __init__(data):
-        _state = _JCrypto.Hash.SHAKE128()
+        _state = _JCrypto.Hash.SHAKE(128)
         _is_squeezing = False
         if data:
             _state.update(data)
```
Comment (mahmoudimus) [inline larky/src/main/resources/vendor/Crypto/Hash/SHAKE128.star:49]: can we update `larky.mutablestruct()` to include `__name__` and `__class__`

File: larky/src/main/resources/vendor/Crypto/Hash/SHAKE256.star [hunk 4]
```diff
@@ -8,20 +13,39 @@ def SHAKE256_XOF(object):
...
-        """
-        Error %d while instantiating SHAKE256
+    def __init__(data=None):
+        _state = _JCrypto.Hash.SHAKE(256)
+        _is_squeezing = False
+        if data:
+            _state.update(data)
+        return larky.mutablestruct(
+            _state=_state,
+            _is_squeezing=_is_squeezing
```
Comment (mahmoudimus) [inline larky/src/main/resources/vendor/Crypto/Hash/SHAKE256.star:23]: can we update `larky.mutablestruct()` to include `__name__` and `__class__`

**Pull request merged**